src/Illuminate/Console/Command.php

```diff
@@ -189,7 +198,94 @@ public function run(InputInterface $input, OutputInterface $output)
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        return $this->laravel->call([$this, 'handle']);
+        if ($this->hasValidInput()) {
+            return $this->laravel->call([$this, 'handle']);
+        }
+
+        return $this->displayValidationErrors();
+    }
+
+    /**
+     * Check if the argument and option values are valid.
+     *
+     * @return bool
+     */
+    protected function hasValidInput()
+    {
+        if (empty($rules = $this->rules())) {
+            return true;
+        }
+
+        try {
+            $this->validator = $this->validator ?? $this->makeValidator();
+        } catch (BindingResolutionException $e) {
+            return true;
```

Comment:
this should not be true here since `$container->make` throws that exception. If so, there is not validators instances.

Reply:
True is being returned here exactly because there is no validator factory bound in the container. This ensures that `illuminate/console` works as before this PR if there is no validator. Otherwise it means that a validator implementation is required to run any command.

Reply:
I don't think this is what we want. If a command has defined rules, then it MUST be validated, non-optionally. It cannot just fail silently.